src/azul/service/responseobjects/buffer.py

```diff
@@ -0,0 +1,55 @@
+from logging import getLogger
+from io import BytesIO
+from typing import Callable
+
+logger = getLogger(__name__)
+
+
+class FlushableBuffer(BytesIO):
+    """
+    A buffer that flushes the output to a callback function (``inbetween_callback``),
+    either if there exists the size of the fragment is large enough
+    (over ``chunk_size``) on flush or when the buffer is closed.
+
+    :param chunk_size: The minimum size of buffer (byte length)
+    :param inbetween_callback: The callback function to receive flushed output
+    """
+
+    def __init__(self, chunk_size: int, inbetween_callback: Callable):
+        super(FlushableBuffer, self).__init__()
+        self.__chunk_size = chunk_size
+        self.__inbetween_callback = inbetween_callback
```

Comment:
This name is annoyingly long. `callback` is fine as this is the only callback in the entire class.

Reply:
Done